app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
+          #add the owner back to the member_ids array
+          mi_array.unshift(@application.owner.owner_ids[0])
+          @application.owner.member_ids = mi_array if @application.owner.member_ids != mi_array
+        end
+        return true
+      else
+        @application.errors.add(:owner, 'Member Ids must be a space separated list of integers')
+        return false
+      end
+    end
+
+    def valid_member_ids(member_ids)
+      if member_ids.nil? || member_ids.empty?
+        return true
+      end
+      !!member_ids.match(NUMBERS_ONLY_REGEX)
+    end
+
+    def set_member_ids
```

Comment:
When I read this method name I read it as saying "I'm setting some instance variable called `@member_ids`".  But the method as written isn't setting anything, it is returning an ID. (while also deleting an entry from an application's `owner.member_ids` array which I don't quite follow?).  So two options:

1. Use a more descriptive name like `get_current_member_ids`, or
2. Since the only place this method is used you are actually setting the value of `@member_ids`, i.e. the calls are currently `@member_ids = set_member_ids`, you could actually set the `@member_ids` variable in this method.  Instance variables are available across methods and in the view, so this is legit.  Probably would still be a good idea to rename the method to something more descriptive.

Reply:
Method updated